src/test/java/lessonfour/hometask/service/impl/userserviceimpl/RegisterUserTest.java

```diff
@@ -0,0 +1,173 @@
+package lessonfour.hometask.service.impl.userserviceimpl;
+
+import lessonfour.hometask.dto.UserRegistrationDto;
+import lessonfour.hometask.dto.UserResponseDto;
+import lessonfour.hometask.exception.DatabaseException;
+import lessonfour.hometask.exception.UserRegistrationDtoException;
+import lessonfour.hometask.model.User;
+import lessonfour.hometask.repository.UserRepository;
+import lessonfour.hometask.service.impl.UserServiceImpl;
+import lessonfour.hometask.validation.UserValidator;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@DisplayName("Unit tests for UserServiceImpl class registerUser method")
+@ExtendWith(MockitoExtension.class)
+class RegisterUserTest {
```

Comment:
Interesting solution to create a separate test class per use case. No need to fix, you can leave it as is